pika/adapters/select_connection.py

```diff
@@ -189,24 +162,27 @@ class SelectPoller(object):
     to override the update_handler and start methods for additional types.
 
     """
-    TIMEOUT = 1
+    TIMEOUT = 5
+    # if the poller used MS specify 1000
+    TIMEOUT_MULT = 1
 
-    def __init__(self, fileno, handler, events, state_manager):
+    def __init__(self, state_manager):
         """Create an instance of the SelectPoller
 
-        :param int fileno: The file descriptor to check events for
-        :param method handler: What is called when an event happens
-        :param int events: The events to look for
         :param method state_manager: The method to manage state
 
         """
-        self.fileno = fileno
-        self.events = events
-        self.open = True
-        self._handler = handler
-        self._timeouts = []
+        self.filenos = dict()
+        self.r_fds = set()
+        self.w_fds = set()
+        self.e_fds = set()
+        self.stopping = False
+        self._timeouts = {}
+        self._next_timeout = None
+        r_interupt, self.w_interupt = os.pipe()
```

Comment:
According to python doc, `select.select()` on Windows only works with sockets; see: https://docs.python.org/2/library/select.html

> On Windows, the underlying select() function is provided by the WinSock library, and does not handle file descriptors that don’t originate from WinSock

So, you will need to create a socket pair instead of a pipe for this in order to be compatible with all supported OS's. But again, there is a caveat: you can't use `socket.socketpair()` for this on Windows. `rabbitpy` solved it like this using its own `_socketpair()` method: https://github.com/gmr/rabbitpy/commit/314924c9485195be20129ecfe2710980e7a01a10#diff-278cd36141e045c14fba20692cbdd05fR460